src/registry.rs

```diff
@@ -251,129 +70,36 @@ impl<'gc> Registry<'gc> {
 
     /// "Stash" a value with a `'gc` branding lifetime, creating a `'static` handle to it.
     ///
-    /// This is a wrapper around an internal `gc-arena::DynamicRootSet` that makes it a little
+    /// This is a wrapper around an internal `gc_arena::DynamicRootSet` that makes it a little
     /// simpler to work with common piccolo types without having to manually specify `Rootable`
     /// projections.
     ///
     /// It can be implemented for external types by implementing the `Stashable` trait.
+    ///
+    /// Values stashed in the global registry are not designed to be held within the Lua state!
+    /// They are 'static handles, not garbage collected values, and thus they are not *traced* like
+    /// garbage collected values to enable cycle collection.
+    ///
+    /// Any value stashed in the global registry will never be freed as long as:
+    ///   1) The global registry itself is alive, and...
+    ///   2) The returned stashed handle is not dropped.
```

Comment:
I'm not sure how to phrase this better, but it's not immediately clear from this that dropping the stashed handle will lead to the value being eventually freed, even if the registry is still alive.  Maybe something along the lines of "[a stashed value] will never be freed as long as both the global registry and the stashed handle are alive."?
(Removing the double negative on the stashed handle)